src/utils/wallet/providers/tomo_wallet.ts

```
@@ -0,0 +1,83 @@
+import { getAddressBalance, getFundingUTXOs, getNetworkFees, getTipHeight, pushTx } from "@/utils/mempool_api";
+import { Fees, Network, UTXO, WalletProvider } from "../wallet_provider";
+
+export const tomoProvider = "tomo_btc";
+
+export class TomoWallet extends WalletProvider{
+
+  constructor() {
+    super();
+  }
+
+  connectWallet = async (): Promise<this> => {
+    if (!window[tomoProvider]) {
+      throw new Error("Tomo Wallet extension not found");
+    }
+
+    await window[tomoProvider].connectWallet()
```

can you please add error handling here and in other methods? things could go wrong, we need to throw a proper error message.

---

Error handling is added